### ISSUE TITLE
Easy way to save to pdf instead of preview?

### ISSUE THREAD

**Issue by sailikk** [iamcco/markdown-preview.nvim#87]
Right now when `:MarkdownPreview`, it opens up a webpage in localhost, and removes it when I switch buffers in Vim. What I would really like is to have several different .md files open in my browser at the same time. Could there be some support such that we can do 

`:MarkdownSave filename.pdf` or bind the .md file to some name `filename.pdf` (maybe the file we are editing is filename.md) so that when we do `:w`, it is saved immediately?

For example,
we're editing file `eg.txt`
We do `MarkdownSave` and it saves and opens up a pdf file saved on your working directory.
we make changes and we get the updated file immediately?

**Comment by blakat360** [comment]
Has there been any progress on this?

**Comment by danjenson** [comment]
also curious

**Comment by wzw-gh** [comment]
u can print the webpage as a pdf file

**Comment by physicophilic** [comment]
pandoc with wkhtmltopdf engine can help do this. Might be slow though.

**Comment by gergap** [comment]
I think you can easily achieve this using *pandoc*. You could create your own `:MarkdownSave` function, which calls *pandoc*. But even better is just writing a Makefile like this:

```
all: title.pdf

view: title.pdf
        xdg-open $<

%.pdf: %.md
        pandoc -o $@ $<
```

Then all you need to do use type `:make<cr>` (I use F7 mapping for this) and `:make view` to generate the PDF and open it in your PDF viewer.

**Comment by juanMarinero** [comment]
### TLDR
A. Remove next string from the downloaded `HTML`
`<div id="__next"><main data-theme="light"><div id="page-ctn" contenteditable="false">`
B. Or replace with `<main data-theme="dark">`
C. `<div id="page-ctn" contenteditable="false">` can be kept

### Details

The downloaded HTML has initially (indented):

```javascript
<!DOCTYPE html>
<html class=" urzbutid idc0_350">
<head>
  <meta http-equiv="content-type" content="text/html; charset=UTF-8">
  [...]
  <link rel="preload" href="markdown_basename_whatever/main-[...].js" as="script">
</head>
<body>
  <div id="__next">
  <main data-theme="light">
  <div id="page-ctn" contenteditable="false">
  <header id="page-header">
    <h3>[...]</h3>
  </header>
  <section class="markdown-body">
```

By **try-and-error** I got the previous [TLDR](#TLDR) workaround. Coded (*zsh*, ~*bash*~):
```sh
x_markdown2pdf_vim_markdown_preview_plugin() {
  cat << "EOF"
Steps:
1. Open the markdown file in (Neo)Vim
2. Run: `:MarkdownPreview`
   Read https://github.com/iamcco/markdown-preview.nvim
3. In browser click "Save Page As..."
4. Terminal go to dir of the downloaded HTML and run
   cd <path_of_downloaded_HTML> # edit ⚠️❗
   <cmds below to edit the HTM>
5. Open in browser
6. Check all the links properly redirect from header-index
   Avoid tildes for ID-links
   Markdown-linter: replace `<` with `&lt;` and `>` with `&gt;`
EOF

  # [ -f "${XDG_CONFIG_HOME:-$HOME/.config}/<x_confirm_PWD__path>" ] && source "${XDG_CONFIG_HOME:-$HOME/.config}/<x_confirm_PWD__path>" # edit ⚠️❗
  x_confirm_PWD
  [[ "$answer_PWD" == "n" ]] && return
  echo "Let's go"

  diff_program="diff"
  diff_program="icdiff" # sudo apt install icdiff # https://github.com/jeffkaufman/icdiff

  # Get the 1st HTML file
  html_files=(*.html)
  html_file=$html_files[1]
  echo "$html_file"

  # Backup
  html_file_original="${html_file::-5}_original_noGit.html" # Remove ".html" + add "_original_NoGit.html"
  echo "$html_file_original"
  cp -v "$html_file" "$html_file_original"

  # Remove <div> for local HTML
  string_to_remove='<div id="__next">'
  echo ""
  echo "Remove: $string_to_remove"
  perl -i -pe 'if (!$done && s/\Q'"$string_to_remove"'\E//) { $done = 1 }' "$html_file" # remove 1st match
  "$diff_program" "$html_file" "$html_file_original" # show changes

  # Optional dark mode
  theme_light='<main data-theme="light">'
  theme_dark='<main data-theme="dark">'
  echo ""
  echo "$theme_light --> $theme_dark"
  perl -i -pe 'if (!$done && s/\Q'"$theme_light"'\E/'"$theme_dark"'/) { $done = 1 }' "$html_file" # replace 1st match
  "$diff_program" "$html_file" "$html_file_original" # show changes

  # Optional remove <div> to extend horizontal space
  string_to_remove='<div id="page-ctn" contenteditable="false">'
  echo ""
  echo "Remove: $string_to_remove"
  perl -i -pe 'if (!$done && s/\Q'"$string_to_remove"'\E//) { $done = 1 }' "$html_file" # remove 1st match
  "$diff_program" "$html_file" "$html_file_original" # show changes

  # Open in browser
  firefox --new-tab "$html_file"
}
x_confirm_PWD() {
  local original_pwd="$PWD"
  echo "PWD is \e[40m $PWD \e[0m"
  unset answer_PWD && echo -n "Do you really want to run this command here (y/N) (Default to 'N'):  "
  read answer_PWD
  answer_PWD=${answer_PWD:-n} # Default to 'n' if no input is provided
  echo
  if [[ "$answer_PWD" == "n" ]]; then
    return
  fi
}
```
